# adele_control settings -----------------------------

# Settings for ros_control control loop
adele_control_loop:
    loop_hz: 200
    cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface
hardware_interface:
    joints:
        - Joint_1
        - Joint_2
        - Joint_3
        - Joint_4
        - Joint_5
        - Joint_6
    # sim_control_mode: 0 # 0: position, 1: velocity
  
    #bonus actuator list for transmission loading
    actuators:
        - Joint_1_motor
        - Joint_2_motor
        - Joint_3_motor
        - Joint_4_motor
        - Joint_5_motor
        - Joint_6_motor
  
# Publish all Joint states ----------------------------
# Creates the /Joint_states topic necessary in ROS
joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

# Joint Trajectory Controller -----------------------------
# For detailed explanations of parameter see http://wiki.ros.org/Joint_trajectory_controller
position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    # Just copy the Joint list from hardware_interface lol
    joints:
        - Joint_1
        - Joint_2
        - Joint_3
        - Joint_4
        - Joint_5
        - Joint_6
    constraints:
        goal_time: 5.0
        #stopped_position_tolerance: 0.4 # Defaults to 0.01
        Joint_1:
            trajectory: 0.60
            goal: 0.15
        Joint_2:
            trajectory: 0.60
            goal: 0.15
        Joint_3:
            trajectory: 0.60
            goal: 0.15
        Joint_4:
            trajectory: 0.60
            goal: 0.15
        Joint_5:
            trajectory: 0.60
            goal: 0.15
        Joint_6:
            trajectory: 0.60
            goal: 0.15
   # gains: # only necessary for velocity and effort controllers
   #     Joint1: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
   #     Joint2: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}

    # state_publish_rate:  50 # Defaults to 50
    # action_monitor_rate: 20 # Defaults to 20
    # hold_trajectory_duration: 0 # Defaults to 0.5
    

   
